src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.CompilationTrackerState.cs

```diff
@@ -167,18 +167,9 @@ private abstract class CompilationTrackerState
                     // DeclarationState now. We'll pass false for generatedDocumentsAreFinal because this is being called
                     // if our referenced projects are changing, so we'll have to rerun to consume changes.
                     return intermediateProjects.Length == 0
-                        ? new AllSyntaxTreesParsedState(solutionServices, compilation, generatorInfo.WithDocumentsAreFinal(false))
+                        ? new AllSyntaxTreesParsedState(compilation, generatorInfo.WithDocumentsAreFinal(false))
                         : new InProgressState(compilation, generatorInfo, compilationWithGeneratedDocuments, intermediateProjects);
                 }
-
-                public static ValueSource<Optional<Compilation>> CreateValueSource(
-                    Compilation compilation,
-                    SolutionServices services)
-                {
-                    return services.SupportsCachingRecoverableObjects && !services.Workspace.Options.GetOption(WorkspaceConfigurationOptions.DisableCompilationTrackerWeakCompilationReferences)
-                        ? new WeakValueSource<Compilation>(compilation)
-                        : new ConstantValueSource<Optional<Compilation>>(compilation);
-                }
```

Comment:
effectively DisableCompilationTrackerWeakCompilationReferences becomes 'true'.  so this entire method just became `return new ConstantValueSource...`.  so that got inlined at all callsites.